### PR TITLE
Improvised shell crafting recipe

### PR DESCRIPTION
**Pull request by FungiFellow**
## About the PR

I added a Recipe for crafting Improvised Shotgun Shells, Recipe is as follows: 1 Steel, 1 Plastic, 4 Glass Shards, and 6 Matches
## Why / Balance

It was added because prior there was no way to obtain ammo for your Improvised Shotgun. There was already a Shell implemented I just added the recipe to craft it.
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
image
## Breaking changes

**Changelog**

:cl:
- add: Added Improvised Shotgun Shell Recipe

### CODE REVIEW

**Comment by tutoumi** [comment]
this recipy is so biiiiiiiiiiiiig, can't you do like : 1 steel, 1 plastic, 5 glass shard, 8 match sticks. instead of add steel, add plastic, add shard, add shard, add shard, add shard...

**Comment by FungiFellow** [comment]
> this recipy is so biiiiiiiiiiiiig, can't you do like : 1 steel, 1 plastic, 5 glass shard, 8 match sticks. instead of add steel, add plastic, add shard, add shard, add shard, add shard...

Glass shards and matchsticks cannot be stacked, so you have to add them individually. For example the Med-Sec Hud recipe has 2 individual capacitors because they can't be stacked
```
﻿- type: constructionGraph
  id: HudMedSec
  start: start
  graph:
    - node: start
      edges:
        - to: medsecHud
          steps:
            - tag: HudMedical
              name: medical hud
              icon:
                sprite: Clothing/Eyes/Hud/med.rsi
                state: icon
              doAfter: 5
            - tag: HudSecurity
              name: security hud
              icon:
                sprite: Clothing/Eyes/Hud/sec.rsi
                state: icon
              doAfter: 5
            - material: Cable
              amount: 5
              doAfter: 5
            - tag: Radio
              name: radio
              icon:
                sprite: Objects/Devices/communication.rsi
                state: walkietalkie
              doAfter: 5
            - tag: CapacitorStockPart
              name: capacitor
              icon:
                sprite: Objects/Misc/stock_parts.rsi
                state: capacitor
              doAfter: 5
            - tag: CapacitorStockPart
              name: capacitor
              icon:
                sprite: Objects/Misc/stock_parts.rsi
                state: capacitor
              doAfter: 5
    - node: medsecHud
      entity: ClothingEyesHudMedSec
```

**Comment by Brandon-Huu** [comment]
Is there a chance for the gun to explode when using these?

**Comment by FungiFellow** [comment]
I found the cause of the Crashing that's kinda dumb
```
id: ShardGlass
 tags:
      - GlassShard
```
THE ID AND TAG AREN'T IDENTICAL.

**Comment by FungiFellow** [comment]
> Is there a chance for the gun to explode when using these?

Nope, just a very inaccurate shell (45 Degree Spread over the 15 degree of regular shells)

**Comment by FungiFellow** [comment]
> Glass shards and matchsticks cannot be stacked

Later on I intend to make Matches able to be butchered into a powder and just use a stack of that instead of 6 individual matches.

But thats for another time

**Comment by Dutch-VanDerLinde** [comment]
Your changelog be broken you need the CL emoji

**Comment by FungiFellow** [comment]
> Your changelog be broken you need the CL emoji

Fixed

**Comment by SpeltIncorrectyl** [comment]
How does it stack to normal buckshot? The same but more inaccurate?

**Comment by FungiFellow** [comment]
> How does it stack up to normal buckshot? The same but more inaccurate?

Same total damage(If you land all pellets), more spread, more pellets but each pellets only does 3/3 slash and pierce.

I performed range tests and effectiveness drops significantly beyond 2.5ish tiles

File: Resources/Prototypes/Recipes/Crafting/Graphs/improvised/improvised_shotgun_shell.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+              icon:
+                sprite: Objects/Tools/matches.rsi
+                state: match_unlit
+              doAfter: 0.5 
+            - tag: Matchstick
+              name: match stick
+              icon:
+                sprite: Objects/Tools/matches.rsi
+                state: match_unlit
+              doAfter: 0.5
```
Comment (metalgearsloth) [inline Resources/Prototypes/Recipes/Crafting/Graphs/improvised/improvised_shotgun_shell.yml:74]:
```suggestion
              doAfter: 0.5
```

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**